crates/ruff_python_formatter/src/comments/visitor.rs

```diff
@@ -17,31 +18,55 @@ use crate::comments::node_key::NodeRefEqualityKey;
 use crate::comments::placement::place_comment;
 use crate::comments::{CommentLinePosition, CommentsMap, SourceComment};
 
+/// Assign comments to nodes through an AST visitor and [`place_comment`].
+pub(super) fn get_comments_map<'a>(
+    root: &'a Mod,
+    source_code: SourceCode<'a>,
+    comment_ranges: &'a CommentRanges,
+) -> CommentsMap<'a> {
+    let mut builder = CommentsBuilder::default();
+    CommentsVisitor::new(source_code, comment_ranges, &mut builder).visit(root);
+    builder.finish()
```

Comment:
Nit: I'm leaning towards inlining this function into `Comments::from_ast` because `get_comments_map` sounds like your getting a value rather than computing it. Maybe rename it to `build_comments_map`? IMO, `CommentsVisitor` and `CommentsBuilder` being visible to the comments module seems reasonable to me.